compute_sdk/globus_compute_sdk/sdk/utils/__init__.py

```diff
@@ -69,3 +71,27 @@ def check_version(task_details: dict | None) -> str | None:
                 f"(worker SDK version: {worker_sdk}; worker OS: {worker_os})"
             )
     return None
+
+
+@t.overload
+def get_env_var_with_deprecation(
+    current_name: str, old_name: str, default: None = None
+) -> str | None: ...
+
+
+@t.overload
+def get_env_var_with_deprecation(
+    current_name: str, old_name: str, default: str
+) -> str: ...
+
+
+def get_env_var_with_deprecation(
+    current_name: str, old_name: str, default: str | None = None
+) -> str | None:
```

Comment:
Verbose, and a shame that mypy can't be smarter than this, but clever and precise.  I'll take it.